src/common/s3.connection.hh

```diff
@@ -19,45 +20,94 @@ class StaticProvider;
 namespace acquire::sink::zarr::common {
 struct S3Connection final
 {
-    S3Connection(const S3Connection&) = delete;
+    S3Connection() = delete;
```

Comment:
I don’t think this is necessary. The default constructor should be deleted automatically if you have a user-defined constructor.